### PR TITLE
ARCBridge header

### PR DESCRIPTION
**Pull request by MaddTheSane**
Include and use the ARCBridge.h header: This makes the code a bit cleaner to read, without the ifdefs all over.

The only place I didn't replace the ifdefs were in the dealloc function, as there isn't that much need to replace those.

### CODE REVIEW

File: Framework/MASPreferencesWindowController.m [hunk 1]
```diff
@@ -39,10 +40,7 @@ - (id)initWithViewControllers:(NSArray *)viewControllers title:(NSString *)title
     NSString *nibPath = [[NSBundle bundleForClass:MASPreferencesWindowController.class] pathForResource:@"MASPreferencesWindow" ofType:@"nib"];
     if ((self = [super initWithWindowNibPath:nibPath owner:self]))
     {
-		_viewControllers = [NSMutableArray arrayWithArray: viewControllers];
-#if !__has_feature(objc_arc)
-        _viewControllers = [_viewControllers retain];
-#endif
+		_viewControllers = RETAINOBJ([NSMutableArray arrayWithArray: viewControllers]);
```
Comment (sorbits) [inline Framework/MASPreferencesWindowController.m]: I suggest we just change it to: `_viewControllers = [viewControllers mutableCopy];`.

File: Framework/MASPreferencesWindowController.m [hunk 2]
```diff
@@ -176,10 +174,7 @@ - (NSToolbarItem *)toolbar:(NSToolbar * __unused)toolbar itemForItemIdentifier:(
         toolbarItem.target = self;
         toolbarItem.action = @selector(toolbarItemDidClick:);
     }
-#if !__has_feature(objc_arc)
-    [toolbarItem autorelease];
-#endif
-    return toolbarItem;
+    return AUTORELEASEOBJ(toolbarItem);
```
Comment (sorbits) [inline Framework/MASPreferencesWindowController.m]: I know this was already like this in the original code, but I think we should call `AUTORELEASEOBJ` in the `alloc/initWithItemIdentifier:` line.

File: Framework/MASPreferencesWindowController.m (hunk 1 above)
Comment (shpakovski) [inline Framework/MASPreferencesWindowController.m]: Perfect, thanks!

**P. S.** Alan, does TextMate still need MRC? Maybe we should just switch to ARC if possible 😃

Comment (sorbits) [inline Framework/MASPreferencesWindowController.m]: > does TextMate still need MRC?

It does not and have not done for ages 😄  so I would definitely not mind dropping the MRC code.

Comment (shpakovski) [inline Framework/MASPreferencesWindowController.m]: I will take care, thanks! 👍

**Comment by sorbits** [comment]
@shpakovski Shall I just remove the MRC code instead of merging this PR?

Maybe we should add a list of “projects using MASPreferences” to this repository.

**Comment by shpakovski** [comment]
I am working on PR right now…

Would be nice to have a reference indeed!

**Pull request closed without merging**